app/components/primer/details_component.rb

```diff
@@ -20,19 +20,23 @@ class DetailsComponent < Primer::Component
 
     with_content_areas :summary, :body
 
-    def initialize(overlay: OVERLAY_DEFAULT, button: BUTTON_DEFAULT, **kwargs)
-      @button = fetch_or_fallback(BUTTON_OPTIONS, button, BUTTON_DEFAULT)
+    def initialize(overlay: OVERLAY_DEFAULT, button: BUTTON_DEFAULT, button_props: {}, **kwargs)
```

Comment:
Hmm. What do you think about using a Slot for the button? That would keep us more in line with the React API: https://primer.style/components/Details

Reply:
I wanted to! but that'd be a breaking change (which I'm trying to avoid for now)

Reply:
If you think it is still worth, I can change it to a slot 😉

Reply:
For the sake of this change, I think it's worth it. Would you be willing to make the necessary changes in dotcom?

cc @muan

Reply:
> that'd be a breaking change (which I'm trying to avoid for now)

Won't going from a content area to a slot already be a breaking change?